### DISCUSSION TITLE
can't compile IpOpt with intel_thread or tbb_thread

### DISCUSSION THREAD

**Discussion by cjl88** [coin-or/Ipopt#459] [Q&A] [answered]
When building ThirdParty-Mumps-master following https://coin-or.github.io/Ipopt/INSTALL.html#SYSTEMPACKAGES, I tried 4 cases for threading
1. -lmkl_sequential
2. -lmkl_gnu_thread  -lgomp
3. -lmkl_tbb_thread
4. -lmkl_intel_thread -liomp5

For last 2, I get errors like:
checking for function dsyev_ in -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_tbb_thread -lmkl_core -lpthread -lm -ldl ... no
checking for function dsyev in -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_tbb_thread -lmkl_core -lpthread -lm -ldl ... no
checking for function DSYEV_ in -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_tbb_thread -lmkl_core -lpthread -lm -ldl ... no
checking for function DSYEV in -L/nfs/software/intel/oneapi/mkl/2021.1.1/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_tbb_thread -lmkl_core -lpthread -lm -ldl ... no
configure: error: Could not find dsyev in Lapack

I am currently having convergence issues using #2 gnu_thread as I mentioned in https://github.com/coin-or/Ipopt/discussions/458, so I am trying #3/4.

Please note the same issue will happen for compiling IpOpt itself.

Any suggestions?

Thanks.

**Comment by svigerske** [comment, marked as answer]
config.log should give some hints on why it couldn't find dsyev. Maybe some other lib is missing. Maybe it's `-ltbb`.

**Reply by cjl88** [reply]
Thanks. config.log does suggest the cause is that TBB lib is missing. Hope configure could have said something related to this.
